src/transform/adm0_surround.py

```diff
@@ -0,0 +1,70 @@
+#############################################################################
+# Process Surrounding Countries for GADM AOI.
+#############################################################################
+
+import sys
+import os
+import fiona
+import geopandas as gpd
+import logging
+import datetime
+import pandas as pd
+from jsonschema import validate
+
+from utils.yaml_api import parse_yaml 
+from utils import gadm_gpkg_processing
+
+import pdb
+
+##
+#
+def transform():
+    config = parse_yaml('config.yml')
+    held_gpkg = os.path.join(
+            config['dirs']['raw_data'], config['surrounding']['gadm']['raw'])
+    country_aoi = config['constants']['ISO3']
+    schema_filename = os.path.join(
+            config['dirs']['schemas'], config['surrounding']['schema'])
+    output_filename = os.path.join(
+            config['dirs']['processed_data'], 
+            config['surrounding']['gadm']['processed'])
+    print(f'Reading {held_gpkg}')
+    for layername in fiona.listlayers(held_gpkg):
+        print(f'Reading {layername} into Geopandas. Takes about 2 mins...')
+        print(datetime.datetime.now().strftime("%Y-%m-%d %H:%M:%S"))
+        gdf = gpd.read_file(held_gpkg, layer=layername)
+        print(r'Done reading.')
+        print(datetime.datetime.now().strftime("%Y-%m-%d %H:%M:%S"))
+    # Process the country of interest 
+    gdf_aoi_levels = gadm_gpkg_processing.get_country_admin_levels(
+            gdf, country_aoi)
+    # Get neighbours
+    neighbours = gadm_gpkg_processing.get_neighbour_countries(
+            gdf, gdf_A0=gdf_aoi_levels['a0'])
+    # Remove country of interest from neighbours list
+    if country_aoi in neighbours: neighbours.remove(country_aoi)
```

Comment:
When this file is created during a normal data scramble, I believe that it also contains the affected country. Do you think there are consequences for excluding it? Should it be added back for now?

Reply:
Not sure. I figured as the Country A0 is already present via another task/process in a separate dataset that the Surrounding Countries dataset is a separate dataset only containing the surrounding countries. It's a single line that does the extraction so easy enough to include/exclude depending on what the requirement is.

Reply:
I was thinking about this one a bit more. Maybe better to include it for now because I think it may be needed for e.g. generating the coastline.